### PR TITLE
create header injector infrastructure

### PR DESCRIPTION
**Pull request by mziolekda**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: extractor/src/main/scala/com/digitalasset/extractor/Extractor.scala [hunk 1]
```diff
@@ -214,7 +214,8 @@ class Extractor[T <: Target](config: ExtractorConfig, target: T) {
         config.appId,
         LedgerIdRequirement(ledgerId = "", enabled = false),
         CommandClientConfiguration(1, 1, overrideTtl = true, java.time.Duration.ofSeconds(20L)),
-        sslContext = config.tlsConfig.client
+        sslContext = config.tlsConfig.client,
+        None
```
Comment (gleber-da) [inline extractor/src/main/scala/com/digitalasset/extractor/Extractor.scala]: Add `parameterName = None` here. Usually named arguments at the end of the arguments list are optional arguments and, given that they can be avoided selectively, it is safer and cleaner to always name them.

File: ledger/ledger-api-integration-tests/src/main/scala/com/digitalasset/platform/semantictest/StandaloneSemanticTestRunner.scala [hunk 2]
```diff
@@ -35,7 +35,7 @@ object StandaloneSemanticTestRunner {
 
     println(s"Running ${nScenarios} scenarios against ${config.host}:${config.port}...")
 
-    val ledgerResource = RemoteServerResource(config.host, config.port, None)
+    val ledgerResource = RemoteServerResource(config.host, config.port, None, None)
```
Comment (gleber-da) [inline ledger/ledger-api-integration-tests/src/main/scala/com/digitalasset/platform/semantictest/StandaloneSemanticTestRunner.scala]: FYI, I'm modifying this code too, we'll have merge conflicts :)

**Comment by stefanobaghino-da** [comment]
Looks stale. @mziolekda can you please close this and delete the branch if it's not useful anymore?

**Comment by stefanobaghino-da** [comment]
> Looks stale. @mziolekda can you please close this and delete the branch if it's not useful anymore?

No reply, closing.

**Pull request closed without merging**